config/parse.py

```diff
@@ -410,18 +411,24 @@ def prefetcher_parse(mod_name, cache):
             ).values()
         )
 
+        listeners = []
```

Comment:
I'm not certain why the listeners are included as part of the configuration? In `main.cc`, you have given the option to the users to select which listeners to activate, so I assume that all listeners are compiled in always. That would make sense, but you add something here to the configuration as if you intend only to compile certain listeners, and not others.

Reply:
I instantiate the listeners in parse.py so I can compile them all in core_inst.cc.inc (see instantiation_file.py:452). Is there a simpler way to do that?

Reply:
Yes, I think a better way is probably to move the `listeners/` directory to `src/listeners/`. This might even work straight out of the box, but you might need to add a line to the makefile.

The reason I raise this concern is because it's not a "module" in the same way that a prefetcher is, where a user might compile one in or not depending.

Reply:
The listeners include both .cc and .h files, so it seems odd to me to put .h files in the src directory. Is that standard practice?

Additionally, aren't the listeners more similar to prefetchers, branch predictors, and replacement policies than the files found in src? The listener directory was modeled after the prefetcher directory in the first place.

Reply:
Put the header files wherever they're findable, in `/include` or in `/src/listeners`. Personally, I prefer the latter, but this repository has traditionally used the former.

Concerning your second question, no, at least not in the sense of ChampSim's interface to the user. Those models are microarchitectural features that get integrated into ChampSim in a special way. Exactly what this way is has reasons for existing that are, again, mostly historical. Nevertheless, for the time being I think that these listeners should be considered internal functionality for now, and public interface once we work out what we mean by "modules". See my recent thoughts [here](https://github.com/ChampSim/ChampSim/issues/424#issuecomment-2353592540).